modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/PagedReflectiveOperationExecutorFactory.java

```diff
@@ -0,0 +1,95 @@
+import org.mule.runtime.module.extension.internal.runtime.connectivity.ExtensionConnectionSupplier;
+import org.mule.runtime.module.extension.internal.runtime.objectbuilder.DefaultObjectBuilder;
+import org.mule.runtime.module.extension.internal.runtime.operation.ReflectiveMethodOperationExecutor;
+import org.mule.runtime.module.extension.internal.runtime.resolver.StaticValueResolver;
+import org.mule.runtime.module.extension.internal.runtime.streaming.PagingProviderProducer;
+import org.mule.runtime.module.extension.internal.util.ReflectionCache;
+
+import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+import javax.inject.Inject;
+
+/**
+ * An implementation of {@link ComponentExecutorFactory} which produces instances of {@link PagedReflectiveOperationExecutorFactory}.
+ *
+ * @param <T> the type of the class in which the implementing method is declared
+ * @since 4.1.6,4.2.0
+ */
+public final class PagedReflectiveOperationExecutorFactory<T, M extends ComponentModel> implements ComponentExecutorFactory<M> {
+
+  private final Class<T> implementationClass;
+  private final Method operationMethod;
+  private final PagingProviderResultTransformer pagingProviderTransformer = new PagingProviderResultTransformer();
+
+  public PagedReflectiveOperationExecutorFactory(Class<T> implementationClass, Method operationMethod) {
+    checkArgument(implementationClass != null, "implementationClass cannot be null");
+    checkArgument(operationMethod != null, "operationMethod cannot be null");
+
+    this.implementationClass = implementationClass;
+    this.operationMethod = operationMethod;
+
+  }
+
+  @Override
+  public ComponentExecutor<M> createExecutor(M operationModel, Map<String, Object> parameters) {
+    DefaultObjectBuilder objectBuilder = new DefaultObjectBuilder(implementationClass, new ReflectionCache());
+    parameters.forEach((k, v) -> objectBuilder.addPropertyResolver(k, new StaticValueResolver<>(v)));
+    Object delegate;
+    CoreEvent initialiserEvent = null;
+    try {
+      initialiserEvent = getInitialiserEvent();
+      delegate = objectBuilder.build(from(initialiserEvent));
+    } catch (Exception e) {
+      throw new MuleRuntimeException(createStaticMessage("Could not create instance of operation class "
+          + implementationClass.getName()), e);
+    } finally {
+      if (initialiserEvent != null) {
+        ((BaseEventContext) initialiserEvent.getContext()).success();
+      }
+    }
+
+    return new ReflectiveMethodOperationExecutor<>(operationModel, operationMethod, delegate, pagingProviderTransformer);
+  }
+
+  private class PagingProviderResultTransformer implements BiFunction<Object, ExecutionContext<M>, Object> {
+
+    @Inject
+    private ExtensionConnectionSupplier connectionSupplier;
+
+    @Override
+    public Object apply(Object result, ExecutionContext<M> executionContext) {
+      PagingProviderProducer pagingProviderProducer =
```

Comment:
you're just moving the ValueTransformers from the mediator here. Why didn't you go with the original map() on the Mono itself that we originally discussed?